build-info-extractor-pip/src/main/java/org/jfrog/build/extractor/pip/extractor/PipBuildInfoExtractor.java

```diff
@@ -0,0 +1,117 @@
+package org.jfrog.build.extractor.pip.extractor;
+
+import org.apache.commons.lang.StringUtils;
+import org.jfrog.build.api.Build;
+import org.jfrog.build.api.Dependency;
+import org.jfrog.build.api.Module;
+import org.jfrog.build.api.builder.DependencyBuilder;
+import org.jfrog.build.api.builder.ModuleBuilder;
+import org.jfrog.build.api.search.AqlSearchResult;
+import org.jfrog.build.api.util.Log;
+import org.jfrog.build.extractor.clientConfiguration.client.ArtifactoryDependenciesClient;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.*;
+
+/**
+ * Created by Bar Belity on 09/07/2020.
+ */
+public class PipBuildInfoExtractor {
+
+    private static final String PIP_AQL_FORMAT =
+            "items.find({" +
+                    "\"repo\": \"%s\"," +
+                    "\"$or\": [{" +
+                    "\"$and\": [{" +
+                    "\"path\": {\"$match\": \"*\"}," +
+                    "\"name\": {\"$match\": \"%s\"}" +
+                    "}]" +
+                    "}]" +
+                    "}).include(\"name\", \"repo\", \"path\", \"actual_sha1\", \"actual_md5\")";
+
+    Build extract(ArtifactoryDependenciesClient client, String repository, String installationLog, Path executionPath, String module, Log logger) throws IOException {
+        // Parse logs and create dependency list of <pkg-name, pkg-file>
+        Map<String, String> downloadedDependencies = PipLogParser.parse(installationLog, logger);
+
+        // Create package-name to dependency map.
+        Map<String, Dependency> dependenciesMap = buildDependenciesMap(downloadedDependencies, client, repository, executionPath, logger);
+
+        // Create Build.
+        List<Dependency> dependenciesList = new ArrayList<>(dependenciesMap.values());
+        return createBuild(dependenciesList, module);
+    }
+
+    Map<String, Dependency> buildDependenciesMap(Map<String, String> downloadedDependencies, ArtifactoryDependenciesClient client, String repository, Path executionPath, Log logger) throws IOException {
```

Comment:
Let's describe the structure of this map (key and value) and also how it is created.

Reply:
Added.